web/src/engine/websites/MangaTown.ts

```diff
@@ -1,13 +1,34 @@
 import { Tags } from '../Tags';
 import icon from './MangaTown.webp';
 import { type Chapter, DecoratableMangaScraper, Page } from '../providers/MangaPlugin';
-import { FetchCSS } from '../platform/FetchProvider';
 import * as Common from './decorators/Common';
 
+//Modded DM5 script
+const pageScript = `
+    new Promise(async (resolve, reject) => {
+        try {
+            const pagecount = window.total_pages;
+            const images = await Promise.all([...new Array(pagecount).keys()].map(async p => {
+                eval(await $.ajax({
+                    url: 'chapterfun.ashx',
+                    data: {
+                        cid: window.chapter_id,
+                        page: p + 1,
+                    }
+                }));
+                return d[0];
```

Comment:
I assume `d[0]` is a global variable which is update within `eval`, therefore all `eval`s should be executed synchronous instead of parallel to avoid race conditions and unexpected `d[0]` assignments.

Reply:
May also check if the call to `eval` return the value that is stored within `d[0]`, then `d[0]` can be avoided at all which automatically solves the issue.

Reply:
I see 👌 . Then perhaps the Dm5 original script should be revised as well according to your specifications.